staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_patch.go

```diff
@@ -54,14 +60,20 @@ func adjustClusterNameIfWildcard(cluster *genericapirequest.Cluster, crdRequest
 
 	var regex *regexp.Regexp
 
-	if cluster.PartialMetadataRequest && crdRequest {
-		regex = crdWildcardPartialMetadataClusterNameRegex
+	if !shard.Wildcard {
+		if cluster.PartialMetadataRequest && crdRequest {
+			regex = crdWildcardPartialMetadataClusterNameRegex
+		} else {
+			regex = wildcardClusterNameRegex
+		}
 	} else {
-		regex = wildcardClusterNameRegex
+		regex = wildcardShardClusterNameRegex
 	}
 
 	matches := regex.FindStringSubmatch(keyWithoutPrefix)
-	if len(matches) >= 2 {
+	if len(matches) >= 3 {
+		return logicalcluster.New(matches[2])
```

Comment:
what does this mean? Some high-level comments would help

Reply:
have a look at
```
	// matches shard/cluster/remainder, capturing shard and cluster.
	//
	// Example: amber/root:org:ws/some-namespace/some-name
	// Example: amber/root:org:ws/some-name
	wildcardShardClusterNameRegex = regexp.MustCompile(`^([^\/]+)\/([^\/]+)\/.+`)
```

Reply:
hard to reason about. Why not move the matching and result logic into the upper if-else construct?

Reply:
done.